.github/workflows/cron-weekly-update-nightly.yml

```diff
@@ -0,0 +1,43 @@
+name: Update Nightly rustc
+on:
+  schedule:
+    - cron: "0 0 * * 0" # runs weekly on Sunday at 00:00
+  workflow_dispatch: # allows manual triggering
+jobs:
+  format:
+    name: Update nightly rustc
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@nightly
+      - name: Update rust.yml to use latest nightly
+        run: |
+          set -x
+          # Not every night has a nightly, so extract the date from whatever
+          # version of the compiler dtolnay/rust-toolchain gives us.
+          NIGHTLY_DATE=$(rustc +nightly --verbose --version | sed -ne 's/^commit-date: //p')
+          # Edit in-place the main CI job for PRs. The only other dependencies
+          # on Nightly in CI are other cronjobs which are fine to use an unpinned
+          # nightly.
+          sed -i "s/toolchain: nightly-.*/toolchain: nightly-${NIGHTLY_DATE}/" .github/workflows/rust.yml || exit 1
+          echo "nightly_date=${NIGHTLY_DATE}" >> $GITHUB_ENV
```

Comment:
A variable would make this much simpler too.

Reply:
In what way? No matter what I need to pull the opaque "nightly" compiler out of dtolnay/rust-toolchain and then query it to learn what its actual version is.

Reply:
Done. It is a bit simpler now in that we don't need to invoke `sed`.